crates/sim2h/src/job/listen.rs

```diff
@@ -12,10 +12,12 @@ impl ListenJob {
         Self { listen, wss_send }
     }
 
-    fn run(&mut self) -> JobContinue {
+    fn run(&mut self) -> JobResult {
         match self.listen.accept() {
             Ok(wss) => {
                 self.wss_send.f_send(wss);
+                // we got data this time, check again right away
+                return JobResult::default();
             }
             Err(e) if e.would_block() => (),
```

Comment:
I have mixed feelings about using this pattern guard, but it's premature optimization to get much further into it. In OCaml I know this can have a severe effect on the pattern matching speed, for instance. https://wiki.c2.com/?PatternMatching is a nice abstract treatment on the topic.

Reply:
Yeah, the problem is, anything else wouldn't be more performant, I still need to do that if there... if only you could make tri-state results in rust : )